native/src/main/java/io/ballerina/stdlib/grpc/listener/ServerCallHandler.java

```diff
@@ -209,21 +209,23 @@ void onMessageInvoke(ServiceResource resource, Message request, StreamObserver r
                 );
             }
             BMap contentContext;
-            if (signatureParamSize >= 1 && (signatureParams.get(0).getTag() == TypeTags.RECORD_TYPE_TAG) &&
-                    signatureParams.get(signatureParamSize - 1).getName().contains("Stream")) {
-                contentContext = ValueCreator.createRecordValue(resource.getService().getType().getPackage(),
-                        MessageUtils.getContextStreamTypeName(resource.getRpcInputType()), valueMap);
-            } else if (signatureParamSize > 1 && (signatureParams.get(1).getTag() == TypeTags.RECORD_TYPE_TAG) &&
-                    signatureParams.get(signatureParamSize - 1).getName().contains("Stream")) {
-                contentContext = ValueCreator.createRecordValue(resource.getService().getType().getPackage(),
-                        MessageUtils.getContextStreamTypeName(resource.getRpcInputType()), valueMap);
-            } else {
-                contentContext = ValueCreator.createRecordValue(resource.getService().getType().getPackage(),
-                        MessageUtils.getContextTypeName(resource.getRpcInputType()), valueMap);
+            if (signatureParamSize >= 1) {
+                Type inputParameter = signatureParams.get(signatureParamSize - 1);
+                if (signatureParamSize == 1 && (signatureParams.get(0).getTag() == TypeTags.RECORD_TYPE_TAG) &&
+                        inputParameter.getName().contains("Stream")) {
+                    contentContext = ValueCreator.createRecordValue(inputParameter.getPackage(),
+                            MessageUtils.getContextStreamTypeName(resource.getRpcInputType()), valueMap);
+                } else if (signatureParamSize > 1 && (signatureParams.get(1).getTag() == TypeTags.RECORD_TYPE_TAG) &&
+                        inputParameter.getName().contains("Stream")) {
+                    contentContext = ValueCreator.createRecordValue(inputParameter.getPackage(),
+                            MessageUtils.getContextStreamTypeName(resource.getRpcInputType()), valueMap);
```

Comment:
```suggestion
                if (((signatureParamSize == 1 && signatureParams.get(0).getTag() == TypeTags.RECORD_TYPE_TAG) || (signatureParamSize > 1 && signatureParams.get(1).getTag() == TypeTags.RECORD_TYPE_TAG)) &&
                        inputParameter.getName().contains("Stream"))) {
                    contentContext = ValueCreator.createRecordValue(inputParameter.getPackage(),
                            MessageUtils.getContextStreamTypeName(resource.getRpcInputType()), valueMap);
                } 
```

Reply:
Shall we combine `if` and `else if`

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/344/commits/ae693d6e69b0bdb882d9e19d380b21090979a3aa